### ISSUE TITLE
Download fails with invalid file system characters

### ISSUE THREAD

**Issue by kylepmanuel** [davidchalifoux/kaput-cli#58]
Attempted to download a file using new -r recursive.

```
thread 'main' panicked at src/put/files.rs:202:64:
creating directory: Os { code: 22, kind: InvalidInput, message: "Invalid argument" }
```

Simple workaround, rename the file/directory in put.io.  In my case, it was ':'

**Comment by davidchalifoux** [comment]
Oh I see, It's because there was an illegal character in the folder name.
To fix this we need to check for and replace illegal characters in file names.

**Comment by davidchalifoux** [comment]
This should be fixed now in v2.4.2.
Common illegal characters are now replaced by default, but this can be disabled with the `--no-replace` flag.

Let me know if you run into any other cases of this.
